src/test/ui/generator/size-moved-locals.rs

```diff
@@ -71,6 +69,6 @@ fn overlap_x_and_y() -> impl Generator<Yield = (), Return = ()>{
 fn main() {
     assert_eq!(1028, std::mem::size_of_val(&move_before_yield()));
     assert_eq!(1032, std::mem::size_of_val(&move_before_yield_with_noop()));
-    assert_eq!(2056, std::mem::size_of_val(&overlap_move_points()));
+    assert_eq!(1032, std::mem::size_of_val(&overlap_move_points()));
```

Comment:
This should still be 2056 because `second` and `first` cannot overlap (the assignment is a single `llvm.memcpy`). This needs fixing before merging.